### DISCUSSION TITLE
Message Recaptcha with react-hook-form

### DISCUSSION THREAD

**Discussion by LimaBruno** [react-hook-form/react-hook-form#2296] [Q&A]
Hey Guys! How can I show unselected recaptcha message using react-hook-form?

Example:

```
import React from "react"
import { useForm } from "react-hook-form"
import ReCAPTCHA from "react-google-recaptcha"

const IndexPage = () => { 
  const { register, handleSubmit, errors } = useForm()
  
  const onSubmit = (data) => { 
    console.log(JSON.stringify(data))       
  }

  const onChange = (value) => {
    console.log("Recaptcha Value:",value)
  }

  return(
    <>
      <h1>Contact</h1>
        <form method="post" action="/" onSubmit={handleSubmit(onSubmit)}>
          First Name<br />
          <input name="example" ref={register({ required: true })} />
          {errors.example && <small>This field is required</small>}
          <br/>
          Last Name<br />
          <input name="exampleRequired" ref={register({ required: true })} />
          {errors.exampleRequired && <small>This field is required</small>}
          <ReCAPTCHA
            sitekey="6LdSWKgZAAAAAKrrp6GFXldEv0I4vqO3yWrTkuNV"
            onChange={onChange}
          />
        <button type="submit">Send</button>
        <button type="reset">Reset</button>
      </form>
    </>
  )
}

export default IndexPage
```

**Comment by bluebill1049** [comment]
i think you should use Controller with the Recaptcah.

**Reply by LimaBruno** [reply]
I looked at the documentation, but I didn't understand how I can do it. Can you give me an example?
A question, any way to use the "register" ("ref={register({ required: true })}") ?

https://react-hook-form.com/advanced-usage/#ControlledmixedwithUncontrolledComponents

**Reply by LimaBruno** [reply]
Searching the communities, I found this post. It's the same problem that I have.

https://spectrum.chat/react-hook-form/help/how-to-add-validation-in-react-google-recaptcha~647a466a-1830-47f8-b92a-fabdb2e46083

What do you think about solution?

**Reply by bluebill1049** [reply]
do you have a codesandbox i can have a play?

**Reply by LimaBruno** [reply]
Hello! added in the container.
I forgot to say that the project is done in Gatsby

https://codesandbox.io/s/blog-s2i6w?file=/src/components/ContactForm/index.js

**Comment by frequencydesign** [comment]
> Hello! added in the container.
> I forgot to say that the project is done in Gatsby
> 
> https://codesandbox.io/s/blog-s2i6w?file=/src/components/ContactForm/index.js

Tried to help because I just implemented reCaptcha with React Hook Form on Monday, but the codesanbox is giving a couple loading errors, including api.getvero.com connection closed

**Comment by bluebill1049** [comment]
thanks @frequencydesign 🙏

**Comment by LimaBruno** [comment]
Hey @bluebill1049, @frequencydesign, are how you?

I solved the problems, below my container link:
https://codesandbox.io/s/react-recaptcha-google-with-react-hook-form-hsqwl

**Comment by LimaBruno** [comment]
Hi @bluebill1049 

Did you look at my container?
I would appreciate your help

**Reply by bluebill1049** [reply]
```
<ReCAPTCHA
          ref={register({ required: true })}
          sitekey="6LcHZ7IZAAAAAJnmo93vjgv3FzZeubeCqhhTYWd5"
          onChange={onChange}
        />```

can you control this into Controller?

**Reply by bluebill1049** [reply]
I don't think ReCAPTCHA expose `Ref` at all, so it's safer to use Controller.

**Reply by LimaBruno** [reply]
It doesn't work this is bad  

`ref={register({ required: true })}`

**Reply by LimaBruno** [reply]
I don't know how I can controller in the recaptcha

https://react-hook-form.com/api/#Controller

Any ideas for doing that?

**Reply by bluebill1049** [reply]
for Controller use `rules`
